src/test/scala/replpp/ScriptRunnerTest.scala

```diff
@@ -0,0 +1,135 @@
+           |  os.write.append(os.Path("$testOutputPath"), "iwashere-@main-foo")
+           |}
+           |
+           |@main def bar() = {
+           |  os.write.append(os.Path("$testOutputPath"), "iwashere-@main-bar")
+           |}""".stripMargin,
+        adaptConfig = _.copy(command = Some("bar"))
+      )
+    } shouldBe "iwashere-@main-bar"
+  }
+
+  "parameters" in {
+    execTest { testOutputPath =>
+      TestSetup(
+        s"""@main def foo(value: String) = {
+           |  os.write.over(os.Path("$testOutputPath"), value)
+           |}""".stripMargin,
+        adaptConfig = _.copy(params = Map("value" -> "iwashere-parameter"))
+      )
+    } shouldBe "iwashere-parameter"
+  }
+
+  "--predefCode" in {
+    execTest { testOutputPath =>
+      TestSetup(
+        s"""os.write.over(os.Path("$testOutputPath"), bar)""".stripMargin,
+        adaptConfig = _.copy(predefCode = Some("val bar = \"iwashere-predefCode\""))
+      )
+    } shouldBe "iwashere-predefCode"
+  }
+
+  "--predefFiles" in {
+    execTest { testOutputPath =>
+      val predefFile = os.temp()
+      os.write.over(predefFile, "val bar = \"iwashere-predefFile\"")
+      TestSetup(
+        s"""os.write.over(os.Path("$testOutputPath"), bar)""".stripMargin,
+        adaptConfig = _.copy(predefFiles = List(predefFile))
+      )
+    } shouldBe "iwashere-predefFile"
+  }
+
+  "additional dependencies via --dependency" in {
+    execTest { testOutputPath =>
+      TestSetup(
+        s"""val compareResult = versionsort.VersionHelper.compare("1.0", "0.9")
+           |os.write.over(os.Path("$testOutputPath"), "iwashere-dependency-param:" + compareResult)
+           |""".stripMargin,
+        adaptConfig = _.copy(dependencies = Seq("com.michaelpollmeier:versionsort:1.0.7"))
+      )
+    } shouldBe "iwashere-dependency-param:1"
+  }
+
+  "additional dependencies via `//> using lib` in script" in {
+    execTest { testOutputPath =>
+      TestSetup(
+        s"""
+           |//> using lib com.michaelpollmeier:versionsort:1.0.7
+           |val compareResult = versionsort.VersionHelper.compare("1.0", "0.9")
+           |os.write.over(os.Path("$testOutputPath"), "iwashere-dependency-using-script:" + compareResult)
+           |""".stripMargin
+      )
+    } shouldBe "iwashere-dependency-using-script:1"
+  }
+
+  "additional dependencies via `//> using lib` in --predefCode" in {
+    execTest { testOutputPath =>
+      TestSetup(
+        s"""
+           |val compareResult = versionsort.VersionHelper.compare("1.0", "0.9")
+           |os.write.over(os.Path("$testOutputPath"), "iwashere-dependency-using-predefCode:" + compareResult)
+           |""".stripMargin,
+        adaptConfig = _.copy(predefCode = Some("//> using lib com.michaelpollmeier:versionsort:1.0.7"))
+      )
+    } shouldBe "iwashere-dependency-using-predefCode:1"
+  }
+
+  "additional dependencies via `//> using lib` in --predefFiles" in {
+    execTest { testOutputPath =>
+      val predefFile = os.temp()
```

Comment:
I passionately hate these JVM APIs that rely on delete-on-exit, dumping all over my `/tmp` for extended periods of time.

Is there nothing similarly convenient that implements `AutoCloseable` ?

Reply:
:) I hear you - unfortunately there isn't anything in [os-lib](https://github.com/com-lihaoyi/os-lib)

[better-files has something like that](https://github.com/pathikrit/better-files#temporary-files), however it's not maintained any longer :(

Reply:
we could add something quite easily ourselves, but for the tests of such a low level project, I'm not convinced that it's worth it
In a different repo, maybe.